lib/async.js

```diff
@@ -46,69 +46,56 @@ const timeoutify = (promise, msec) =>
     );
   });
 
-const throttle = (timeout, fn, ...args) => {
-  let timer;
-  let wait = false;
-
-  const execute = args
-    ? (...pars) => (pars ? fn(...args, ...pars) : fn(...args))
-    : (...pars) => (pars ? fn(...pars) : fn());
-
-  const delayed = (...pars) => {
-    timer = undefined;
-    if (wait) execute(...pars);
-  };
-
-  const throttled = (...pars) => {
-    if (!timer) {
-      timer = setTimeout(delayed, timeout, ...pars);
-      wait = false;
-      execute(...pars);
+const throttle = (fn, interval, ...presetArgs) => {
+  if (typeof interval !== 'number' || interval < 0) {
+    throw new Error('Interval must be greater then 0');
+  }
+  let lastTime = 0;
+  return (...args) => {
+    const now = Date.now();
+    if (now - lastTime < interval) return;
+    try {
+      fn(...presetArgs, ...args);
+    } finally {
+      lastTime = now;
     }
-    wait = true;
   };
-
-  return throttled;
 };
 
-const debounce = (timeout, fn, ...args) => {
-  let timer;
-
-  const debounced = () => (args ? fn(...args) : fn());
-
-  const wrapped = () => {
+const debounce = (fn, delay, ...args) => {
+  let timer = null;
+  return () => {
     if (timer) clearTimeout(timer);
-    timer = setTimeout(debounced, timeout);
+    timer = setTimeout(() => {
+      fn(...args);
+      timer = null;
+    }, delay);
   };
-
-  return wrapped;
 };
 
 const callbackify =
-  (fn) =>
+  (asyncFn) =>
   (...args) => {
     const callback = args.pop();
-    fn(...args)
-      .then((value) => {
-        callback(null, value);
-      })
-      .catch((reason) => {
-        callback(reason);
-      });
+    if (typeof callback !== 'function' || callback.length !== 2) {
+      throw new Error('Last argument should be a function with 2 parameters');
+    }
```

Comment:
We do not need such a check, callback may handle just first argument `error` and will be thrown while it is ok